src/main/java/org/stellar/sdk/requests/SSEStream.java

```diff
@@ -83,13 +99,30 @@ public void run() {
             }
           }
         });
+
+    // Start a timer to check if the client has not received any event for a long time.
+    // If so, we will close the connection and restart it.
+    latestEventTime.set(System.currentTimeMillis());
+    clientTimeoutTimer.scheduleAtFixedRate(
```

Comment:
rather than another thread, can this be run in the start thread loop - https://github.com/stellar/java-stellar-sdk/pull/555/files#diff-e2dabf286e824384ee81a568eccc0deaea2558961523521b93f048abb37792baR77

Reply:
Thanks! changed.